src/components/DependencyGraph.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { useEffect, useMemo, useState } from 'react';
+import { css } from 'emotion';
+import { LoaderComponent } from './LoaderComponent';
+import {
+    DependencyLink,
+    DependencyNode,
+    Environment,
+    EnvironmentStateDto,
+    Network,
+    Service,
+    ServiceDto,
+} from './types';
+import { createLinks, mapServiceDtoToService } from '../helpers/MappingHelpers';
+import { EnvironmentSelect } from './EnvironmentSelect';
+import { CheckboxProps, Radio } from 'semantic-ui-react';
+import { Graph } from './Graph';
+
+export const DependencyGraph: React.FC = () => {
+    const [nodes, setNodes] = useState<DependencyNode[]>([]);
+    const [isPending, setIsPending] = useState<boolean>(true);
+    const [services, setServices] = useState<Service[]>([]);
+    const [env, setEnv] = useState<Environment>(Environment.SIT);
+    const [radio, setRadio] = useState<boolean | undefined>(false);
+
+    const network: Network = useMemo(() => {
+        const nodes = services.map((service: Service) => ({
+            name: service.name,
+            version: service.version,
+        }));
+        return {
+            nodes,
+            links: createLinks(nodes, services),
+        };
+    }, [services]);
+
+    const onlyNodesWithContracts = useMemo(() => {
+        return network.nodes.filter(
+            (node: DependencyNode) =>
+                network.links.findIndex(
+                    (link: DependencyLink) =>
+                        (link.source.name === node.name &&
+                            link.source.version === node.version) ||
+                        (link.target.name === node.name &&
+                            link.target.version === node.version)
+                ) >= 0
+        );
+    }, [network]);
+
+    useEffect(() => {
+        setIsPending(true);
+        fetch(
```

Comment:
Would be good to move all http requests to a separate module. Also, axios would be a better choice than plain fetch